src/test/fuzz/chainstate.cpp

```diff
@@ -0,0 +1,693 @@
+    std::unordered_map<COutPoint, CTxOut, SaltedOutpointHasher> utxos;
+
+    //const auto first_time{SteadyClock::now()};
+
+    // On every round start from a freshly copied initial datadir.
+    fs::remove_all(g_test_data.working_datadir);
+    fs::copy(g_test_data.init_datadir, g_test_data.working_datadir, fs::copy_options::overwrite_existing | fs::copy_options::recursive);
+
+    // Create the chainstate..
+    uint64_t prune_target{0};
+    if (fuzzed_data_provider.ConsumeBool()) {
+        prune_target = fuzzed_data_provider.ConsumeIntegral<uint64_t>();
+    }
+    node::BlockManager::Options blockman_opts{
+        .chainparams = chainparams,
+        .prune_target = prune_target,
+        .blocks_dir = g_test_data.working_datadir / "blocks",
+        .notifications = g_test_data.notifs,
+    };
+    const ChainstateManager::Options chainman_opts{
+        .chainparams = chainparams,
+        .datadir = g_test_data.working_datadir,
+        // TODO: make it possible to call CheckBlockIndex() without having set it here, and call it in CallOneOf().
+        .check_block_index = true,
+        .checkpoints_enabled = false,
+        .minimum_chain_work = UintToArith256(uint256{}),
+        .assumed_valid_block = uint256{},
+        .notifications = g_test_data.notifs,
+        .signals = &g_test_data.main_signals,
+    };
+    ChainstateManager chainman{*g_setup->m_node.shutdown, chainman_opts, blockman_opts};
+
+    // ..And then load it.
+    node::CacheSizes cache_sizes;
+    cache_sizes.block_tree_db = 2 << 20;
+    cache_sizes.coins_db = 2 << 22;
+    cache_sizes.coins = (450 << 20) - (2 << 20) - (2 << 22);
+    node::ChainstateLoadOptions load_opts {
+        .prune = prune_target > 0,
+        .require_full_verification = false,
+        .coins_error_cb = nullptr,
+    };
+    auto [status, _] = node::LoadChainstate(chainman, cache_sizes, load_opts);
+    assert(status == node::ChainstateLoadStatus::SUCCESS);
+
+    //const auto time_before_loop{SteadyClock::now()};
+
+    BlockValidationState dummy_valstate;
+    std::vector<CBlock> blocks_in_flight;
+    LIMITED_WHILE(fuzzed_data_provider.ConsumeBool(), 10'000) {
+        // Every so often, update our cache used to create non-coinbase txs.
+        if (_count % 100 == 0) AppendUtxos(chainman, utxos);
+
+        CallOneOf(fuzzed_data_provider,
+            // Process a list of headers. Most of the time make it process the header of a valid block
+            // cached for future processing.
+            [&]() NO_THREAD_SAFETY_ANALYSIS {
+                LOCK(cs_main);
+                std::vector<CBlockHeader> headers;
+
+                // In 1% of the cases, generate a random list of headers to be processed. Otherwise, create a single
+                // valid block.
+                // TODO: make it possible to generate a chain of more than one valid block.
+                const bool is_random{fuzzed_data_provider.ConsumeIntegralInRange(0, 99) == 99};
```

Comment:
When debating this with myself my weak argument in favour of it was that the target was very slow and this could help discover interesting paths faster. I also wanted your feedback on this, so thanks.

Reply:
Done.